Update get started with human task orchestration to include Play (#4401)
camunda/camunda-docs: docs/guides/getting-started-orchestrate-human-tasks.md

import OperateHumanTasks from './img/operate-human-tasks.png';
import FormEditorImg from './img/form-editor.png';
import NavigationHistoryImg from './img/modeler-navigation-history.png';
import PlayButtonImg from './img/play-play-button.png';
import PlayOpenFormImg from './img/play-open-form.png';
import PlayChickenImg from './img/play-chicken-complete.png';

import clsx from "clsx";
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import SaasPrereqs from './react-components/saas-prerequisites.md'
import Install from './react-components/install-c8run.md'

> You can use Camunda 8 to orchestrate processes with human tasks of any complexity. User tasks allow you to create and assign tasks to users, who can then perform their work and enter the required data to drive the business process.

Reviewer: ```suggestion
You can use Camunda 8 to orchestrate processes with human tasks of any complexity. Use the user task BPMN element to create and assign tasks to users, who can then perform their work and enter the required data to drive the business process.
```
For the getting started with HTO guide, I'd also be interested in seeing @volodymyr-melnykc 's perspective as he's the HTO PM

import PlayButtonImg from './img/play-play-button.png';
import PlayOpenFormImg from './img/play-open-form.png';
import PlayChickenImg from './img/play-chicken-complete.png';
import HumanTasksImg from './img/human-task-bpmn-diagram-tasks.png';
import HumanTasksGatewayImg from './img/human-task-bpmn-diagram-gateway.png';
import PlayOperateImg from './img/play-operate.png';
import OperateTasklistImg from './img/operate-open-tasklist.png';

import clsx from "clsx";
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';
import SaasPrereqs from './react-components/saas-prerequisites.md'
import Install from './react-components/install-c8run.md'

You can use Camunda 8 to orchestrate processes with human tasks of any complexity. Use the [user task BPMN element](/components/modeler/bpmn/user-tasks/user-tasks.md) to create and assign tasks to users, who can then perform their work and enter the required data to drive the business process.

## About this guide

This guide introduces you to the basics of human task orchestration. You will create a simple process to decide on what to eat for dinner, and drive the process flow according to that decision.

> - Select **Self-Managed** to see an example using Camunda 8 Run and Desktop Modeler.

Author: Thanks, I understand now, appreciate the clarification - I have changed to Desktop and Web (to match the download language), and updated the intro:

![image](https://github.com/user-attachments/assets/8930d4d8-6699-4e8c-b2c7-a16bbb3770ba)

This guide introduces you to the basics of human task orchestration. You will create a simple process to decide on what to eat for dinner, and drive the process flow according to that decision.

- Select **Desktop** to see an example using Camunda 8 Run and Desktop Modeler in a local development environment.

> - Select **Web** to see an example using Camunda 8 SaaS and Web Modeler.

Reviewer: Web Modeler Self-Managed is excluded, but that's appropriate for this stage of the journey

<TabItem value="saas">

You will learn how to:

- Create and run your first process with a human in the loop.
- Build a BPMN diagram using Web Modeler.
- Create and link a form to a user task.
- Validate your process using Play mode.

<p><img src={HumanTaskDiagramImg} alt="The completed BPMN diagram" /></p>

</TabItem>
<TabItem value="sm">

You will learn how to:

- Create and run your first process with a human in the loop.
- Build a BPMN diagram using Modeler.
- Create and link a form to a user task.
- Deploy and run your process.
- Complete the task in Tasklist, and check the process in Operate.

</TabItem>
</Tabs>

Perform the following steps to create and run your first process with a human in the loop.

## Step 1: Create a new process

Start by designing a process to demonstrate how to route the process flow based on a user decision. In this example, you will create a process to decide what to eat for dinner.

### Create a new file

<Tabs groupId="install" className="tabs-hidden">
<TabItem value="saas">

First, you must create the BPMN diagram file and project.

1. Open Web Modeler, and click **New project**. Every file in Web Modeler requires a project.

> 2. Select **Create new > BPMN diagram**.

Reviewer: For automation, users should always use process applications. They have the most features, including a smoother Play deployment experience

</TabItem>
</Tabs>

<Tabs groupId="install" className="tabs-hidden">
<TabItem value="saas">

In this example, you will learn how to:

- Create and run your first process with a human in the loop.
- Build a BPMN diagram using [Web Modeler](/components/modeler/web-modeler/launch-web-modeler.md).
- Create and link a form to a [user task](/components/modeler/bpmn/user-tasks/user-tasks.md).
- Validate your process using [Play mode](/components/modeler/web-modeler/play-your-process.md).

<p><img src={HumanTaskDiagramImg} alt="The completed BPMN diagram." /></p>

</TabItem>
<TabItem value="sm">

You will learn how to:

- Create and run your first process with a human in the loop.
- Build a BPMN diagram using Modeler.
- Create and link a form to a user task.
- Deploy and run your process.
- Complete the task in Tasklist, and check the process in Operate.

</TabItem>
</Tabs>

Perform the following steps to create and run your first process with a human in the loop.

## Step 1: Create and design a process

> Start by creating and designing a process to demonstrate how to route the process flow based on a user decision. In this example, you will create a process to decide what to eat for dinner.

Reviewer 2: ```suggestion
To learn how to route a process flow based on a user decision, you will create and design a process to decide what to eat for dinner.
```

> 1. A **start event** is automatically added to the canvas. Click it to display configuration and append options.

Reviewer: The start event should be automatically selected- you can check this behavior on SaaS

> 2. Click the rectangular **Append Task** icon to append a task.

Reviewer: <img width="806" alt="image" src="https://github.com/user-attachments/assets/f0d29da9-95ba-4212-9c21-dfc79c921919">
These instructions should be updated to account for the new context pad. The user has to hover over the blue circle to the right, then click the rectangular append task icon. This applies for all instructions- they should be double-checked

</TabItem>
</Tabs>

1. A **start event** is automatically added to the canvas and selected.

> 2. Append a task. For example, hover over the blue plus icon and select the **Append Task** icon from the [context pad](/components/modeler/web-modeler/context-pad.md).

Reviewer 2: Context pad is more useful for the SaaS/web modeler side of things and not the same experience in desktop. If this section is swapped to `Append element` instead (and the context pad reference removed), it brings it back in line with the Desktop experience and would remove the need for another set of split/tabbed content.

## Step 2: Design a form

You have now designed the process. To allow the user to make the decision, you must next design a [form](../components/modeler/forms/camunda-forms-reference.md). Forms are added to user tasks and start events to capture user input, with the user input used to route the process flow, make API requests, or orchestrate your services.

<Tabs groupId="install" className="tabs-hidden">
<TabItem value="saas">

1. Select the user task you created in **[Step 1](#step-1-create-a-new-process)**.

> 1. Click the blue **link icon** in the lower right corner. A menu expands that allows you to create a new form.

Reviewer: <img width="432" alt="image" src="https://github.com/user-attachments/assets/dae13207-660c-471f-b82f-2f5d40ad61cc">
I saw this instead of the empty state

<!-- TODO note that processes can be of any complexity, and link to advanced guides -->

## Step 2: Design and link a form

You have now designed the process. To allow the user to make the decision, you must next design a [form](../components/modeler/forms/camunda-forms-reference.md). Forms are added to user tasks and start events to capture user input that is used to route the process flow, make API requests, or orchestrate your services.

<Tabs groupId="install" className="tabs-hidden">
<TabItem value="saas">

:::note

- You do not need to save your process diagram as Web Modeler automatically saves every change you make.

> - You can also create [milestones](/components/modeler/web-modeler/milestones.md) to save a version snapshot of your BPMN or DMN diagram at any time.

Reviewer: NOTE: Milestones have been renamed to versions within the last few months

1. Click and drag a **Radio** component into the form to create a radio group. Enter a descriptive name in the properties panel.
1. Set a **key** which maps to a process variable. The value of the component is stored in this variable, and can be read by the process that uses this form. As already defined by the conditions in the process earlier, use the variable `meal`.
   <img src={FormValuesTop} style={{width: 250}} alt="Defining a radio group's name and key." />
1. Scroll down to the **Static options** section of the properties panel to add radio options. Since there are two options for dinner, add an extra value by clicking on the plus sign. Enter the value `Chicken` with a label of `Chicken`, and in the other value enter the value `Salad` with a label of `Salad`.
   <img src={FormValuesBottom} style={{width: 250}} alt="Defining a radio group's static option values." />

1. In your form's properties panel, copy the Form ID for use in your process.

   <img src={FormId} style={{width: 250}} alt="The form properties panel, showing the form ID." />

Now you have created and designed the form, you must link it to your process.

> 1. Open the process you created in **[Step 1](#step-1-create-a-new-process)** by clicking on the process file's name in the top bar.

Reviewer 2: ```suggestion
1. Open the process you created in **[Step 1](#step-1-create-and-design-a-process)** by clicking on the process file's name in the top bar.
```

2. Select the user task, and open the **Form** menu in the properties panel.

> 3. In the Form menu, enter the **Form ID** for the form you created in **[Step 2](#step-2-design-a-form)**.

Reviewer 2: ```suggestion
3. In the Form menu, enter the **Form ID** for the form you created in **[Step 2](#step-2-design-and-link-a-form)**.
```

<!-- TODO note that processes can be of any complexity, and link to advanced guides -->

## Step 2: Design and link a form

You have now designed the process. To allow the user to make the decision, you must next design a [form](../components/modeler/forms/camunda-forms-reference.md). Forms are added to user tasks and start events to capture user input that is used to route the process flow, make API requests, or orchestrate your services.

<Tabs groupId="install" className="tabs-hidden">
<TabItem value="saas">

:::note

- You do not need to save your process diagram as Web Modeler automatically saves every change you make.
- You can also create [milestones](/components/modeler/web-modeler/milestones.md) to save a version snapshot of your BPMN or DMN diagram at any time.

:::

1. Select the user task you created in **[Step 1](#step-1-create-a-new-process)**.

> 1. Click the blue **Link Form** icon in the context pad, and click **Create new form**.

Reviewer: This is the first time the "context pad" is mentioned, but there's nothing showing what it looks like or where to find it. The same thing applies to the palette. @lmbateman are these intuitive enough concepts that we can assume the user knows what we're talking about?

Author: I've added a link to the context pad page here 👍

> ## Step 3: Link the form to your process

Reviewer: If you create a form according to these instructions, then this step is automatically completed.

Forms linked in the user task are deployed together with the process. If you make changes to a form, you have to deploy the referencing process again to make the changes appear.
:::

<Tabs groupId="install" className="tabs-hidden">
<TabItem value="saas">

## Step 4: Validate your process using Play

You can use [Play mode](/components/modeler/web-modeler/play-your-process.md) to quickly validate the process behavior and play different scenarios.

1. Click the **Play** tab to enter Play mode.

> 1. Once the Play environment is ready, click **Start a process instance** to start testing your process.

Reviewer: With 8.6, the user will need a cluster rather than just waiting for the Play environment.
![image](https://github.com/user-attachments/assets/c9c659e3-9afb-41cf-b7d2-d2deaa640df8)
![image](https://github.com/user-attachments/assets/840778c7-7365-4a94-8c67-5e8575e44746)
![image](https://github.com/user-attachments/assets/145e1d21-4ffa-4cc2-933c-5ea63ac73083)
![image](https://github.com/user-attachments/assets/e6b10b33-6ea3-4834-83c6-d42433029b6c)

   <img src={PlayChickenImg} style={{width: 800}} alt="Completing the Play process" />

   <p>When the execution arrives at this gateway, all outgoing sequence flows are evaluated in the order in which they have been defined. The sequence flow which condition evaluates to ‘true’ is selected for continuing the process.</p>
   <p>In this example, the token moved through the gateway and (according to the conditional expressions outlined earlier) to the selected dinner based on the <code>Decide what's for dinner</code> user task you completed.</p><p>As you chose <code>Chicken</code>, the token moved through to <code>Prepare chicken</code> and successfully completed. If you had selected <code>Salad</code>, the token would have moved through to <code>Prepare salad</code> instead.</p>

1. Click **Rewind** on the canvas and select the `Salad` option to test this flow also works correctly.

:::note

As well as using Play mode to quickly validate and run your process in development, you can also:

- Deploy the process to a [cluster](/components/concepts/clusters.md) in other environments such as testing, staging, and production. After you deploy your process, it can be run on the cluster.

> - Run and complete the user task in [Tasklist](/components/tasklist/introduction-to-tasklist.md). Applications such as Tasklist can be used by humans to complete tasks. As well as using Play mode and Tasklist to run a process, you can call the process via the API or an inbound trigger. Read more about [run options](/components/modeler/web-modeler/run-or-publish-your-process.md).

Reviewer: You can follow a link to Operate...
![image](https://github.com/user-attachments/assets/42c511d3-7748-419a-859c-ac579a640525)
and then a link to Tasklist if you have an open user task...
![image](https://github.com/user-attachments/assets/748418bc-906f-4d0b-82e2-5f8cf6ace7fb)

Or you can open Tasklist and it should show the open task directly

Finally, check that you can complete a user task in Tasklist.

When the process instance arrives at the user task, a new user task instance is created at Zeebe. The process instance stops at this point and waits until the user task is completed. Applications like Tasklist can be used by humans to complete these tasks. In this last step, you will open Tasklist to run the user task you created.

:::tip
While it may originally seem like the goal of automating a process is to remove humans entirely, efficiently allocating work through user tasks can be even more beneficial. Within this example, we've included a form to demonstrate the completion of a user task.

Using the Zeebe or Tasklist API, many other ways to complete a user task are possible, such as redirecting to another application to complete the task, or even listening to IoT devices to capture human interaction with the real world via job workers.
:::

1. Open Tasklist at `http://localhost:8080/tasklist`.

2. On the left, you will notice a list of **tasks**. There should be one open task `Decide what's for dinner`. Click this task to open it in the detail view.

> 3. In the detail view, the form you created in **[Step 2](#step-2-design-a-form)** appears. It is read only since this task is currently unassigned. You have to claim the task to work on it. Next to **Assignee**, click **Assign to me** to claim the task.

Reviewer 2: ```suggestion
3. In the detail view, the form you created in **[Step 2](#step-2-design-and-link-a-form)** appears. It is read only since this task is currently unassigned. You have to claim the task to work on it. Next to **Assignee**, click **Assign to me** to claim the task.
```
